examples/kaleidoscope/implementation_typed_pointers.rs

```diff
@@ -0,0 +1,1200 @@
+use std::borrow::Borrow;
```

Comment:
can be fixed, but using `cfg` inline would get very messy. Fixing it should not be too hard, because I don't think function pointers are used here. So it's mostly loads that needs to specify the types of the value that they are loading, and perhaps some other minor API changes.

Reply:
It's okay if we fix it in a followup PR, but I'd like the example to work in all LLVM versions